# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/quota/ForwardDecls.h
# Commit: f799f35b1c84
# Full Hash: f799f35b1c84a8c16f1cca5e09cda13372dc8528
# Author: Jan Varga <jvarga@mozilla.com>
# Date: 2024-10-10 09:22:33
# Regressor Bug: 1866402
# File Overlap Count: 1
# Description:
#   Bug 1866402 - Change all origin operations which uninit origins to return an array of origin metadata; r=dom-storage-reviewers,jari
#   
#   This is a preparation for unsetting some flags when origins are either cleared
#   or reset.
#   
# ==============================================================================

diff -r 0afeb47f38b0 -r f799f35b1c84 dom/quota/ForwardDecls.h
--- a/dom/quota/ForwardDecls.h	Wed Oct 09 22:16:49 2024 +0000
+++ b/dom/quota/ForwardDecls.h	Wed Oct 09 22:16:50 2024 +0000
@@ -10,6 +10,7 @@
 #include <cstdint>
 #include <functional>
 
+#include "nsTArrayForwardDeclare.h"
 #include "mozilla/dom/quota/CommonMetadataArrayFwd.h"
 #include "mozilla/dom/quota/Config.h"
 
@@ -68,8 +69,12 @@
 using UniversalDirectoryLockPromise =
     MozPromise<RefPtr<UniversalDirectoryLock>, nsresult, true>;
 
+struct OriginMetadata;
+using OriginMetadataArray = nsTArray<OriginMetadata>;
 class UsageInfo;
 
+using OriginMetadataArrayPromise =
+    MozPromise<OriginMetadataArray, nsresult, true>;
 using OriginUsageMetadataArrayPromise =
     MozPromise<OriginUsageMetadataArray, nsresult, true>;
 using UsageInfoPromise = MozPromise<UsageInfo, nsresult, false>;